internal/services/sarif/sarif.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2022 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sarif
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/ZupIT/horusec-devkit/pkg/entities/analysis"
+	"github.com/ZupIT/horusec-devkit/pkg/entities/vulnerability"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/severities"
+
+	"github.com/ZupIT/horusec/cmd/app/version"
+)
+
+type Sarif struct {
+	analysiss *analysis.Analysis
+
+	resultsByTool          map[string][]Result
+	rulesByToolAndID       map[string]map[string]Rule
+	artifactsByToolAndName map[string]map[string]Artifact
+}
+
+func NewSarif(analysiss *analysis.Analysis) *Sarif {
+	return &Sarif{
+		analysiss:              analysiss,
+		resultsByTool:          make(map[string][]Result),
+		rulesByToolAndID:       make(map[string]map[string]Rule),
+		artifactsByToolAndName: make(map[string]map[string]Artifact),
+	}
+}
+
+func (s *Sarif) ConvertVulnerabilityToSarif() (report Report) {
+	report.Runs = []ReportRun{}
+	s.populateReferenceMaps(&report)
+	s.buildReportRun(&report)
+
+	// SARIF output format version/schema
+	report.Version = "2.1.0"
+	report.SchemaURI = "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json"
+
+	return report
+}
+
+// Iterate all vulnerabilities and break them into the maps/associations necessary
```

Comment:
is a good practice in go to start comments with the function name as in the guide @matheusalcantarazup mentioned, please replace with `populateReferenceMaps Iterate all vulnerabilities and break ...`